pvanalytics/system.py

```diff
@@ -1 +1,118 @@
+    """Determine system azimuth and tilt from power or POA.
+
+    Solar noon is estimated on each day by fitting a quadratic to data
+    in `power_or_poa` and finding the vertex of the fit. A brute force
+    search is performed on clearsky POA irradiance for all pairs of
+    candidate azimuths and tilts (`azimuths` and `tilts`) to find the
+    pair that results in the closest azimuth at solar noon to the
+    azimuths calculated from the curve fitting step. Closest is
+    determined by minimizing the sum of squared error between the
+    azimuth at solar noon on each day in `power_or_poa` and the
+    azimuth at solar noon (maximum POA) in the clearsky POA
+    irradiance.
+
+    The accuracy of the values returned from this function will vary
+    with the time-resolution of the clearsky and solar position
+    data. For the best accuracy pass `solar_azimuth`, `solar_zenith`,
+    and the clearsky data (`ghi`, `dhi`, and `dni`) with one-minute
+    timestamp spacing. If `solar_azimuth` has timestamp spacing less
+    than one minute it will be resampled and interpolated to estimate
+    azimuth at each minute of the day. Regardless of the timestamp
+    spacing these parameters must cover the same days as
+    `power_or_poa`.
+
+    Parameters
+    ----------
+    power_or_poa : Series
+        Timezone localized series of power or POA irradiance
+        measurements.
+    sunny : Series
+        Boolean series with True for values when it is sunny.
+    tilts : list of floats
+        List of candidate tilts.
+    azimuths : list of floats
+        List of candidate azimuths.
+    solar_azimuth : Series
+        Time series of solar azimuth.
+    solar_zenith : Series
+        Time series of solar zenith.
+    ghi : Series
+        Clear sky GHI.
+    dhi : Series
+        Clear sky DHI.
+    dni : Series
+        Clear sky DNI.
+
+    Returns
+    -------
+    azimuth : float
+    tilt : float
+
+    Notes
+    -----
+    Based on PVFleets QA project.
+
+    """
+    peak_times = _peak_times(power_or_poa[daytime])
+    azimuth_by_minute = solar_azimuth.resample('T').interpolate(
+        method='linear'
+    )
+    modeled_azimuth = azimuth_by_minute[peak_times]
+    best_azimuth = None
+    best_tilt = None
+    smallest_sse = None
+    for azimuth in azimuths:
+        for tilt in tilts:
+            poa = pvlib.irradiance.get_total_irradiance(
+                tilt,
+                azimuth,
+                solar_zenith,
+                solar_azimuth,
+                ghi=ghi,
+                dhi=dhi,
+                dni=dni
+            ).poa_global
+            poa_azimuths = azimuth_by_minute[
+                _group.by_day(poa).idxmax()
+            ]
+            filtered_azimuths = poa_azimuths[np.isin(
+                poa_azimuths.index.date,
+                modeled_azimuth.index.date
+            )]
+            sum_of_squares = sum(
+                (filtered_azimuths.values - modeled_azimuth.values)**2
+            )
```

Comment:
This won't actually work since the `modeled_azimuth` series includes the same dates, but potentially different times from `poa_azimuths`. After trying a few things out I think it is best to leave it as is.